src/drivers/intel/cavs/ipc.c

```diff
@@ -231,7 +231,6 @@ int ipc_platform_send_msg(struct ipc_msg *msg)
 		goto out;
```

Comment:
would also be nice to replace the above with `return -EBUSY;` and change the end of the function to `return 0;` and remove the `ret` variable to help make it clear that only when 0 is returned, we need to remove the message from the list. Same for Renoir, Baytrail, Haswell and iMX. Others already do it that way.

Reply:
good point! I made a commit for it.